it/uk/gov/hmrc/agentmapping/MappingControllerISpec.scala

```diff
@@ -9,490 +9,212 @@ import play.api.libs.ws.WSClient
+      val splitFixtures: Seq[(Seq[TestFixture], Seq[TestFixture])] =
+        (1 until Math.max(5, fixtures.size)).map(fixtures.splitAt) ++ (1 until Math.max(4, fixtures.size))
+          .map(fixtures.reverse.splitAt)
+
+      splitFixtures.foreach {
+        case (left, right) =>
+          val leftTag = left.map(f => Service.asString(f.service)).mkString(",")
+          s"return created when we add all, but some mappings already exist: $leftTag" in {
+            givenIndividualRegistrationExists(utr)
+            givenUserIsAuthorisedForMultiple(left)
+            createMappingRequest().putEmpty().status shouldBe 201
+
+            givenUserIsAuthorisedForMultiple(fixtures)
+            createMappingRequest().putEmpty().status shouldBe 201
+
+            verifyKnownFactsCheckAuditEventSent(2)
+            fixtures.foreach(f => verifyCreateMappingAuditEventSent(f))
+            verifyCreateMappingAuditEventSent(AgentCodeTestFixture)
+          }
+
+          s"return conflict when all mappings already exist, but we try to add again: $leftTag" in {
+            givenIndividualRegistrationExists(utr)
+            givenUserIsAuthorisedForMultiple(fixtures)
+            createMappingRequest().putEmpty().status shouldBe 201
+
+            givenUserIsAuthorisedForMultiple(left)
+            createMappingRequest().putEmpty().status shouldBe 409
+
+            givenUserIsAuthorisedForMultiple(right)
+            createMappingRequest().putEmpty().status shouldBe 409
+
+            verifyKnownFactsCheckAuditEventSent(3)
+            fixtures.foreach(f => verifyCreateMappingAuditEventSent(f))
+            verifyCreateMappingAuditEventSent(AgentCodeTestFixture)
+          }
+      }
     }
 
     "return forbidden when the supplied arn does not match the DES business partner record arn" in {
-      givenUserAuthorisedFor(`IR-SA-AGENT`, IRAgentReference, "A1111A", "testCredId", agentCodeOpt = Some("TZRXXV"))
+      givenUserIsAuthorisedFor(`IR-SA-AGENT`, IRAgentReference, "A1111A", "testCredId", agentCodeOpt = Some(agentCode))
       givenIndividualRegistrationExists(utr)
 
-      new Resource(s"/agent-mapping/mappings/${utr.value}/TARN0000001/${saOnlyIdentifier}", port).putEmpty().status shouldBe 403
-
-      verifyAuditRequestSent(
-        1,
-        event = AgentMappingEvent.KnownFactsCheck,
-        detail = Map(
-          "knownFactsMatched" -> "false",
-          "utr" -> "2000000000",
-          "agentReferenceNumber" -> "TARN0000001",
-          "authProviderId" -> "testCredId"),
-        tags = Map(
-          "transactionName" -> "known-facts-check",
-          "path" -> "/agent-mapping/mappings/2000000000/TARN0000001/IRAgentReference~A1111A"))
+      new Resource(s"/agent-mapping/mappings/${utr.value}/TARN0000001", port).putEmpty().status shouldBe 403
+
+      verifyKnownFactsCheckAuditEventSent(1, matched = false, arn = "TARN0000001")
     }
 
     "return forbidden when there is no arn on the DES business partner record" in {
```

Comment:
These tests around DES are specific to `IR-SA-AGENT`.  
Should they be non-specific tests run against all enrolments, i.e. moved into the tests on all fixtures above at line 90?

Reply:
These DES tests don't depend on the enrolments held, I would not duplicate them.